crypto/evp/evp_enc.c

```diff
@@ -581,35 +581,37 @@ static int evp_cipher_init_skey_internal(EVP_CIPHER_CTX *ctx,
     if (enc) {
         if (ctx->cipher->einit_skey == NULL) {
             /* Attempt fallback for providers that do not support SKEYs */
-            const unsigned char *keydata;
-            size_t keylen;
+            const unsigned char *keydata = NULL;
+            size_t keylen = 0;
 
-            if (!EVP_SKEY_get_raw_key(skey, &keydata, &keylen)) {
+            if (skey != NULL && !EVP_SKEY_get_raw_key(skey, &keydata, &keylen)) {
```

Comment:
This does not look right to me. skey != NULL should be checked at the beginning of the function.

Reply:
Let me disagree. When we pass the NULL skey, we still may want to set IV (multiple-pass init, as the current interface does)

Reply:
Hmm..., right.

Reply:
No, it's fine. I completely forgot about that use case.